src/propagation.h

```diff
@@ -136,9 +119,20 @@ class SpanContext : public ot::SpanContext {
   uint64_t trace_id_;
   OptionalSamplingPriority propagated_sampling_priority_ = nullptr;
   std::string origin_;
+  std::unordered_map<std::string, std::string> baggage_;
+  // Trace tags are key/value pairs that are propagated along a trace.  If this
+  // `SpanContext` was extracted, then `extracted_trace_tags_` contains any
+  // trace tags parsed from the "x-datadog-trace-tags" header.  If this
+  // `SpanContext` is later injected, these trace tags will be included as the
+  // "x-datadog-trace-tags" header, possibly modified.
+  std::unordered_map<std::string, std::string> extracted_trace_tags_;
+  // `extracted_upstream_services_` contains the parsed value of the
+  // "_dd.p.upstream_services" tag from the trace tags.  If this `SpanContext`
+  // is later injected, then the upstream services will be included as a trace
+  // tag, possibly with this service's information appended to it.
+  std::vector<UpstreamService> extracted_upstream_services_;
```

Comment:
Consider not parsing everything. For one, we don't need most of the data, and also it leaves room for upstream bugs to break us in the future.